ClaudiaRojasSoto/Blog_App
# Feature/add devise

## Add Devise for User Authentication

### Description
This pull request adds the Devise gem to the application to manage user registration and authentication. Devise provides a flexible and secure way to handle user accounts. This ensures that users can register, confirm their emails, and reset their passwords.

### General Requirements
- [x] Used correct Gitflow
- [x] Work is well-documented

### Ruby Requirements
- Followed all best practices for Ruby as per guidelines.

### Project Requirements
- [x] Removed `current_user` method from `ApplicationController` as Devise provides one.
- [x] Installed and set up Devise.
- [x] Users can register a new account.
- [x] Users can log in with a combination of email and password.
- [x] Hashed passwords are being stored in the database.
- [x] Email confirmation upon registration.
- [x] Users can reset their passwords.
- [x] Modified Devise views for registration and login to match the wireframes and styling.

### How to Test
1. Clone the feature branch into your local machine.
2. Run `bundle install` to install the Devise gem.
3. Run `rails db:migrate` to apply the new Devise-related database migrations.
4. Start the Rails server.
5. Go to the registration and login pages to check if they match the wireframes and your styling.

Please, review the changes made on this feature and give me feedback to improve my app. Thank you in advance.

### Contributors
@ClaudiaRojasSoto 
@Dachrono
